Odata-docs/webapi-8/fundamentals/server-driven-paging.md

```diff
@@ -0,0 +1,251 @@
+
+Here's an example of response:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Products",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Product 1" 
+        },
+        {
+            "Id": 2,
+            "Name": "Product 2" 
+        },
+    ],
+    "@odata.nextLink": "http://localhost:5000/odata/Products?$skiptoken=foobar"
+}
+```
+
+In this example, the client would make a request to `GET http://localhost:5000/odata/Products?$skiptoken=foobar` to retrieve the next page.
+
+The format of the next link is opaque to the client. This means that the client should not try to decode the link, modify it or make any assumptions about how it is generated or what its components mean. It can only assume that it's a valid URL that will return the next page. This allows the server freedom in how it chooses to generate and interpret the next link. It allows the server to change the underlying implementation of the next link without affecting the client.
+
+## Configuring paging using `PageSize`
+
+ASP.NET Core OData allow you to configure the page size on the server side using the [`PageSize`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute.pagesize) property of the [`[EnableQuery]`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute) attribute. This will automatically limit the number of items in the response and include an `@odata.nextLink` property.
+
+The following code snippet demonstrates how to configure the `PageSize` on a controller action:
+
+```c#
+public class ProductsController : ODataController
+{
+    [EnableQuery(PageSize = 2)]
+    public IQueryable<Product> Get()
+    {
+        return productsCollection;
+    }
+}
+```
+
+This sets the number of items per page to 2. Let's assume the entity set has a total of 5 products. To fetch the first page, simply fetch the entity set:
+
+```http
+GET http://localhost:5000/Products
+```
+
+**Response**:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Product 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Product 2"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/Products?$skip=2"
```

Comment:
```suggestion
    "@odata.nextLink": "http://localhost:5000/Products?$skiptoken=Id-2"
```

Reply:
WebAPI uses `$skip` in the next link by default if you don't explicitly enable `SkipToken()`. I mention this in the "Improving pagint with `$skiptoken` section:


> You may have noticed that ASP.NET Core OData generates next links using `$skip`. Using `$skip` on large collections may lead to performance degradation in some databases. For more information about the drawbaks of `$skip`, visit the [client-driven paging](/odata/webapi-8/fundamentals/client-driven-paging#drawbacks) article.